### ISSUE TITLE
Improvements for the next year

### ISSUE THREAD

**Issue by msschwartz21** [dlmbl/intro_ml#4]
- [x] Put a checkpoint after the end of the data wrangling before moving on to model content
- [x] Define one hot encoding and make its usage more explicit in the models. Consider having students write a function for converting to one hot encoding.
- [x] Explicitly state usage of epsilon in the softmax function 
- [x] Variable `model-path` in unused in the tensorflow section

Other notes
- The initial linear classifier is very vulnerable to bugs depending on how the students implement the functions they are assigned. Additional docstrings defining inputs/outputs could help, but maybe standalone code tests could help confirm that the student's functions are working correctly

**Comment by msschwartz21** [comment]
- [x] Use mamba
- [x] Write setup.sh
- [x] Add solution flags to exercise
